### ISSUE TITLE
[4.5.4] broken buld with PHP 8

### ISSUE THREAD

**Issue by remicollet** [swoole/swoole-src#3683]
Using 8.0.0beta4

```
/dev/shm/BUILD/php80-php-pecl-swoole4-4.5.4/NTS/swoole_client.cc: In function 'void zim_swoole_client_getSocket(zend_execute_data*, zval*)':
/dev/shm/BUILD/php80-php-pecl-swoole4-4.5.4/NTS/swoole_client.cc:1247:69: error: 'php_sockets_le_socket' was not declared in this scope; did you mean 'php_swoole_get_socket'?
 1247 |     SW_ZEND_REGISTER_RESOURCE(return_value, (void *) socket_object, php_sockets_le_socket());
      |                                                                     ^~~~~~~~~~~~~~~~~~~~~

```

sockets are no more resource but object, and needed stuff have been made public for some other ext, see
http://git.php.net/?p=php-src.git;a=commitdiff;h=4c43806b617e8001d2bfe8c6639640d06ec6e8f4

**Comment by remicollet** [comment]
If it may help, changes in "event"

https://github.com/remicollet/event/commit/7a0b8c79c1dfcdc0ced132685a720b6199f71794#diff-812cd78e57015776b1cfecb917ccf010

**Comment by twose** [comment]
Thank you @remicollet , someone reported this problem before, but I forgot to fix it...
We almost no longer use it, just for backward compatibility, so I will fix it later...

**Comment by remicollet** [comment]
Thanks,
I confirm the fix is OK with 8.0.0beta4
